src/test/groovy/edu/oregonstate/mist/api/CardsResourceTest.groovy

```diff
@@ -0,0 +1,359 @@
+package edu.oregonstate.mist.api
+
+import com.google.common.base.Optional
+import edu.oregonstate.mist.api.jsonapi.ResultObject
+import edu.oregonstate.mist.cardsapi.core.Card
+import edu.oregonstate.mist.cardsapi.db.CardDAO
+import edu.oregonstate.mist.cardsapi.db.CardFluent
+import edu.oregonstate.mist.cardsapi.resources.CardsResource
+import groovy.mock.interceptor.StubFor
+import io.dropwizard.jersey.params.IntParam
+import org.junit.Test
+
+class CardsResourceTest {
+
+    // Test CardsResource.getCards()
+    @Test
+    void testGetCards() {
+        def mockDao = new StubFor(CardDAO)
+        def mockFluent = new StubFor(CardFluent)
+        mockDao.demand.getValidTypes ()
+                { -> ["skill", "attack", "power", "status", "curse"] }
+        mockDao.demand.getValidColors ()
+                { -> ["red", "green", "blue", "colorless"] }
+        mockDao.demand.getValidRarities ()
+                { -> ["basic", "common", "uncommon", "rare"] }
+        def dao = mockDao.proxyInstance()
+        mockFluent.demand.getCards (0..10) {
+            List<String> types, String name, List<String> colors,
+            List<String> rarities, Integer energyMin, Integer energyMax,
+            List<String> keywords, Integer cardNumber, Boolean isRandom -> []
+        }
+        def fluent = mockFluent.proxyInstance()
+        CardsResource resource = new CardsResource(dao, null, fluent,
+                dao.getValidTypes(), dao.getValidColors(), dao.getValidRarities())
+
+        // Test: no result
+        def noResult = resource.getCards(["attack"], Optional.absent(),
+                ["colorless"], ["rare"], Optional.absent(), Optional.absent(),
+                null, Optional.absent(), Optional.absent())
+        assert noResult.status == 200
+        assert noResult.entity.data == []
+
+        // Test: Invalid type
+        def invalidType = resource.getCards(["invalidType"], Optional.absent(),
```

Comment:
You could take advantage of the `with` clause from Groovy to simplify the code: https://dzone.com/articles/getting-groovy-with-with